generators/python/src/fern_python/generators/context/pydantic_generator_context_impl.py

```diff
@@ -53,6 +53,17 @@ def __init__(
             ):
                 self._non_union_self_referencing_type_ids.add(id)
 
+        self._non_union_types_with_self_referencing_dependencies: Dict[str, Set[str]] = dict()
```

Comment:
# This map goes from every non union type to a list of referenced types that circularly reference themselves
nit: Dict[ir_types.TypeId, Set[ir_types.TypeId]]

Reply:
Updated!